devito/ops/transformer.py

```diff
@@ -224,7 +224,7 @@ def create_ops_par_loop(trees, ops_kernel, parameters, block, name_to_ops_dat,
     for tree in trees:
         if isinstance(tree, IterationTree):
             for bounds in [it.bounds() for it in tree]:
-                it_range.extend(bounds)
+                it_range.extend(bounds[:-1]+(bounds[-1]+1,))
```

Comment:
i wonder if there might be a way to access it's upper bound so even if the tuple position eventually changes, we're still referring to the same parameter. 
otherwise, looks good to me

Reply:
Found a solution! Check it out on the last commit =)